kernel/src/syscall/accept.rs

```diff
@@ -18,7 +18,14 @@ pub fn sys_accept(
 ) -> Result<SyscallReturn> {
     debug!("sockfd = {sockfd}, sockaddr_ptr = 0x{sockaddr_ptr:x}, addrlen_ptr = 0x{addrlen_ptr:x}");
 
-    let fd = do_accept(sockfd, sockaddr_ptr, addrlen_ptr, Flags::empty(), ctx)?;
+    let fd =
+        do_accept(sockfd, sockaddr_ptr, addrlen_ptr, Flags::empty(), ctx).map_err(
+            |err| match err.error() {
+                // FIXME: `accept` should not be restarted if a timeout has been set on the socket using `setsockopt`.
+                Errno::EINTR => Error::new(Errno::ERESTARTSYS),
```

Comment:
Move the errno number conversion into `do_accept` so that it won't be duplicated in both `sys_accept` and `sys_accept4`.